rpm/0045-sailfishos-gecko-Add-and-adjust-embedlite-static-pre.patch

```diff
@@ -0,0 +1,152 @@
+
+Adds the various embedlite static preferences to StaticPrefList.yaml so
+that these will be availble for use as static pref variables in the C++
+code.
+
+[sailfishos][gecko] Set security.external_protocol_requires_permission to default to false
+
+Upstream introduces the new user preference:
+
+security.external_protocol_requires_permission
+
+This defaults to true, but we need it to default to false so that
+external URL actions continue to work as intended. Here we change the
+default value of the preference from true to false so as to allow this.
+
+[sailfishos][gecko] Set embedlite.azpc.json.doubletap static pref to default to true
+
+When the preferences were moved to the StaticPrefList.yaml file the
+embedlite.azpc.json.doubletap was set to default to false, but it should
+have been set to true. If set to false, double-tap always zooms, even on
+items that support double-taps.
+
+[sailfishos][gecko] Set gfx.webrender.force-disabled to default to true
+
+Sailfish OS requires the Webrender pipeline to be disabled. Setting
+gfx.webrender.force-disabled on start-up in WebEngineSettings at the app
+level isn't enough, so it needs to be patched to default to true in all
+cases.
+---
+ modules/libpref/init/StaticPrefList.yaml      | 59 ++++++++++++++++++-
+ modules/libpref/init/all.js                   |  2 +-
+ modules/libpref/moz.build                     |  1 +
+ .../handling/ContentDispatchChooser.jsm       |  2 +-
+ 4 files changed, 61 insertions(+), 3 deletions(-)
+
+diff --git a/modules/libpref/init/StaticPrefList.yaml b/modules/libpref/init/StaticPrefList.yaml
+index 058df55a189d..615143df6b5e 100644
+--- a/modules/libpref/init/StaticPrefList.yaml
++++ b/modules/libpref/init/StaticPrefList.yaml
+@@ -3986,6 +3986,63 @@
+   value: false
+   mirror: always
+ 
++#ifdef MOZ_EMBEDLITE
++  #---------------------------------------------------------------------------
++  # Prefs starting with "embedlite."
++  #---------------------------------------------------------------------------
++
++# Default azpc notifications behavior
++-   name: embedlite.azpc.handle.viewport
++    type: bool
++    value: true
++    mirror: always
++
++-   name: embedlite.azpc.handle.singletap
++    type: bool
++    value: false
++    mirror: always
++
++-   name: embedlite.azpc.handle.longtap
++    type: bool
++    value: false
++    mirror: always
++
++-   name: embedlite.azpc.json.singletap
++    type: bool
++    value: true
++    mirror: always
++
++-   name: embedlite.azpc.json.doubletap
++    type: bool
++    value: true
++    mirror: always
++
++-   name: embedlite.azpc.json.longtap
++    type: bool
++    value: true
++    mirror: always
++
++-   name: embedlite.azpc.json.scroll
++    type: bool
++    value: false
++    mirror: always
++
++# Make gecko compositor use GL context/surface provided by the application
++-   name: embedlite.compositor.external_gl_context
++    type: bool
++    value: true
```

Comment:
I see browser as a special case. Thus, default should be false and browser set this to true.

https://github.com/sailfishos/sailfish-components-webview/pull/169/commits/234f548ff61f721cb06b4c28ebe86235870d87d9#r1698347125

Reply:
Okay; I switched this to `false` and it's now set in the browser initialisation code. Is this what you were thinking?

Reply:
Yeap, defaulting to false was exactly what I was thinking about.

Reply:
It's perfectly like this as well so that sailfish-components-webview and sailfish-browser do alter prefs. Let's create more PRs on top as needed.